test/langtools/tools/javac/sealed/PrivateMembersInPermitClause.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8338981
+ * @summary Access to private classes should be permitted inside the permits clause of the enclosing top-level class
+ * @library /tools/lib
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ *                     jdk.compiler/com.sun.tools.javac.main
+ *                     jdk.compiler/com.sun.tools.javac.util
+ * @build toolbox.ToolBox toolbox.JavacTask toolbox.Task
+ * @run main PrivateMembersInPermitClause
+ */
+
+import java.nio.file.Path;
+import java.util.Objects;
+import toolbox.Task;
+import java.util.List;
+
+public class PrivateMembersInPermitClause extends toolbox.TestRunner {
+
+    private final toolbox.ToolBox tb;
+
+    public PrivateMembersInPermitClause() {
+        super(System.err);
+        tb = new toolbox.ToolBox();
+    }
+
+    public static void main(String... args) throws Exception {
+        new PrivateMembersInPermitClause().runTests();
+    }
+
+    public void runTests() throws Exception {
+        runTests(_ -> new Object[] {});
+    }
+
+    @Test
+    public void givenPrivateClassInPermitsClause_whenCompiling_thenShouldCompile() throws Exception {
```

Comment:
I would use shorter method names, you can probably add more details about what the test does using javadoc or an internal comment

Reply:
it may be the way, but comments could become danglig over time, not sure if its the way. If I were writing with junit, I would use @DisplayName, not sure if there is something like this in jtreg. I will se what can be done though